imap_processing/codice/tests/test_decom.py

```diff
@@ -94,3 +221,21 @@ def test_ways_to_get_data(decom_test_data):
     data_value_using_list = decom_test_data[0][1]
     # Check if data is same
     assert data_value_using_key == data_value_using_list
+
+
+def test_unit_conversion(validation_data, filtered_analog):
+    """Test if unit conversion is correct"""
+    # Call the function to perform unit conversion on the validation data
+    engineering_data = validate_unit_conversion(validation_data, filtered_analog)
+
+    # Assert that both DataFrames have the same shape
+    assert engineering_data.shape == validation_data.shape
+
+    # Define a tolerance for floating-point precision
+    tolerance = 1e-6  # Adjust this tolerance as needed
```

Comment:
This can be deleted. Put it there as an "option" if we need something like this.

Reply:
I think you might be able to do a numpy array comparison with allclose for your tolerance:
https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html#numpy.testing.assert_allclose

If you don't need a tolerance you should remove it though so we don't accidentally introduce some math slop somewhere.

Reply:
I don't think it is needed. I was just thinking about the decimal places because they aren't whole numbers. Def do not want math slop. I will look into the links.